lib/cloud/azure/imds.go

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package azure
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/gravitational/trace"
+)
+
+const (
+	// metadataReadLimit is the largest number of bytes that will be read from imds responses.
+	metadataReadLimit = 1_000_000
```

Comment:
Is there any reason why we need 1kk here, then can you move it to `constants.go` and use for both implementations, otherwise I think you can just re-use https://github.com/gravitational/teleport/blob/c219962cccdef534d640f321804874c6bc876425/constants.go#L501-L509